line_profiler/_line_profiler.pyx

```diff
@@ -128,9 +145,39 @@ cdef class LineProfiler:
             import warnings
             warnings.warn("Could not extract a code object for the object %r" % (func,))
             return
-        if code not in self.code_map:
-            self.code_map[code] = {}
-            self.functions.append(func)
+
+        if code.co_code in self.dupes_map:
+            self.dupes_map[code.co_code] += [code]
+            # code hash already exists, so there must be a duplicate function. add no-op
+            co_code = code.co_code + (9).to_bytes(1, byteorder=byteorder) * (len(self.dupes_map[code.co_code]))
+            CodeType = type(code)
+            
+            if hasattr(code, 'replace'):
+                # python 3.8+
+                code = func.__code__.replace(co_code=co_code)
+            else:
+                # python <3.8
+                co = code
+                code = CodeType(co.co_argcount, co.co_kwonlyargcount,
```

Comment:
Maybe use a inline utility function? `compat_code_replace`? You could use Cython `IF` preprocessor directives to eliminate the branch and `hasattr` call at compile time.

Reply:
IF directives are deprecated sadly, and I don't believe I can call hasattr at compile time. I did consider a utility function, but this only happens once iirc.